lib/patterns/hooks.coffee

```diff
@@ -0,0 +1,55 @@
+class Hooks
```

Comment:
Make it the engine behind Functions before, around, after as they are going to be the wrapper for it